lib/segment/src/id_tracker/id_tracker_base.rs

```diff
@@ -287,6 +293,22 @@ impl IdTracker for IdTrackerEnum {
         }
     }
 
+    fn make_immutable(&self, save_path: &Path) -> OperationResult<IdTrackerEnum> {
+        match self {
+            IdTrackerEnum::MutableIdTracker(id_tracker) => id_tracker.make_immutable(save_path),
+            IdTrackerEnum::ImmutableIdTracker(id_tracker) => {
+                Ok(id_tracker.make_immutable(save_path)?)
+            }
+        }
+    }
```

Comment:
Instead of `make_immutable` on all implementations, I think I'd prefer `ImmutableIdTracker::from(other_id_tracker)`.

What do you think?

Reply:
I think this is not possible. [Here](https://github.com/qdrant/qdrant/blob/dd326e9091bf18db73230344ee3cd998818f8614/lib/collection/src/collection_manager/optimizers/segment_optimizer.rs#L380) we only have access to the `dyn IdTracker` trait, not the simple id tracker value. This is why I added this conversion function on trait level.